modules/services/gpg-agent.nix

```diff
@@ -87,13 +88,9 @@ in
           "default-cache-ttl-ssh ${toString cfg.defaultCacheTtlSsh}"
       );
 
-      home.sessionVariables =
-        optionalAttrs cfg.enableSshSupport {
-          SSH_AUTH_SOCK = "\${XDG_RUNTIME_DIR}/gnupg/S.gpg-agent.ssh";
-        };
-
       programs.bash.initExtra = gpgInitStr;
       programs.zsh.initExtra = gpgInitStr;
+      xsession.initExtra = gpgInitStr;
```

Comment:
Should be in `xsession.profileExtra` and I guess the `SSH_AUTH_SOCK` also should be imported into the systemd environment.